solr/core/src/test/org/apache/solr/handler/component/SearchHandlerTest.java

```diff
@@ -360,6 +360,19 @@ public void testLuceneIOExceptionHandling() throws Exception {
       assertTrue(
           e.getMessage()
               .contains("Query contains too many nested clauses; maxClauseCount is set to 1"));
+
+      solrQuery = new SolrQuery("{!surround maxBasicQueries=1000 df=title}10W(tes*,test*)");
+      solrQuery.add(ShardParams.SHARDS_TOLERANT, "true");
+      final QueryRequest req4 = new QueryRequest(solrQuery);
+      req4.setMethod(SolrRequest.METHOD.POST);
+      e =
+          assertThrows(
+              BaseHttpSolrClient.RemoteSolrException.class,
+              () -> req4.process(cloudSolrClient, collectionName));
+      assertEquals(400, e.code());
+      assertTrue(
+          e.getMessage()
+              .contains("Query contains too many nested clauses; maxClauseCount is set to 1"));
```

Comment:
curious: with a mini solr cluster of size 1, wouldn't this request have failed even before this change?

Reply:
Even with the cluster of size 1, it still flows through the distributed query section of the SearchHandler and previously did not result in an error, but would just return no results and partial results as true.

Reply:
oh, nice! so it was _very_ tolerant before 😁